### PR TITLE
local file sink

### PR DESCRIPTION
**Pull request by tomas-quix**
Only JSON tested.

Issue: https://github.com/quixio/quix-streams/issues/559

### CODE REVIEW

File: quixstreams/sinks/community/file/__init__.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+_UNSAFE_CHARACTERS_REGEX = re.compile(r"[^a-zA-Z0-9 ._]")
+
+
+class InvalidFormatError(Exception):
+    """
+    Raised when the format is specified incorrectly.
+    """
+
+
+class FileSink(BatchingSink):
```
Comment (tomas-quix) [inline quixstreams/sinks/community/file/__init__.py]: Is it normal to keep class in init file? I would not expect it here.

File: quixstreams/sinks/community/file/__init__.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+        If `append` is `True` and an existing file is found, data will be appended to
+        the last file. Otherwise, a new file is created based on the batch's starting
+        offset.
+
+        :param batch: The batch of data to write.
+        """
+
+        # Generate directory based on topic and partition
+        directory = Path(self._output_dir)
```
Comment (tomas-quix) [inline quixstreams/sinks/community/file/__init__.py]: This is something that should be in separate method so it is easy to overload.

File: quixstreams/sinks/community/file/__init__.py (hunk 1 above)
Comment (gwaramadze) [reply quixstreams/sinks/community/file/__init__.py]: There is technically nothing wrong with it, but I will move it to file.py and import the class in __init__.py

File: quixstreams/sinks/community/file/sink.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        If `append` is `True` and an existing file is found, data will be appended to
+        the last file. Otherwise, a new file is created based on the batch's starting
+        offset.
+
+        :param batch: The batch of data to write.
+        """
+
+        # Generate directory based on topic and partition
+        # Generate filename based on a padded offset
+        # Returns existing file path if append is True
```
Comment (ghost) [inline quixstreams/sinks/community/file/sink.py]: Looks like the comment refers to something different here

Comment (gwaramadze) [reply quixstreams/sinks/community/file/sink.py]: Yeah the wording is poor here. I will skip this comment since there is a big on in the method itself.

**Review by tim-quix — DISMISSED** [review]
Tested using my File Source Connector, works :+1:

**Comment by tim-quix** [comment]
One last thing we may wanna add is a docs page before merging

**Comment by gwaramadze** [comment]
Docs page added: https://quixdocsdev.blob.core.windows.net/pr432/quix-streams/connectors/sinks/file-sink.html

**Review by ghost — APPROVED** [review]

**Pull request merged**